labs/sysdev-accessing-hardware-stm32mp2/sysdev-accessing-hardware-stm32mp2.tex

```diff
@@ -0,0 +1,756 @@
+collide with future changes made by the maintainers for this file.
+
+A more futureproof idea is to create a new Device Tree file which
+includes the standard one, and adds custom definitions. So, create a
+new \code{arch/arm64/boot/dts/st/stm32mp257f-dk-custom.dts} file containing:
+
+\begin{verbatim}
+/dts-v1/;
+#include "stm32mp257f-dk.dts"
+&i2c8 {
+            status = "okay";
+            /delete-property/ pinctrl-names;
+};
+\end{verbatim}
+
+As you can see, it's also possible to include \code{dts} files, and not
+only \code{dtsi} ones.
+
+Why the \code{/delete-property/} statement? That's because we want
+to see what happens when a device doesn't have associated pin
+definitions yet.
+
+A device like an I2C controller node is typically declared in the DTSI
+files for the SoC, without pin settings as these are board
+specific. Pin definitions are then usually defined at board level.
+
+In our case, we don't see such definitions, but they are actually
+found in the \kfile{arch/arm64/boot/dts/st/stm32mp25xx-dkx.dtsi}
+file, shared between multiple stm32mp25 DK boards, which is included
+by the toplevel Device Tree for our board.
+
+Modify the \kfile{arch/arm64/boot/dts/st/Makefile} file to add your custom
+Device Tree, and then have it compiled (\code{make dtbs}).
+
+Reboot your board with the update.
+
+Back to the running system, we can now see that there is one more
+I2C bus. We can also recognize the I2C8 address (\code{0x0000000046040000})
+though it's now associated to the \code{i2c-0} device name, which
+already existed previously, but mapped to the I2C2 controller.
+
+\begin{bashinput}
+# i2cdetect -l 
+i2c-1	i2c       	STM32F7 I2C(0x0000000046040000) 	I2C adapter
+i2c-0	i2c       	STM32F7 I2C(0x0000000040130000) 	I2C adapter
+\end{bashinput}
+
+Pay attention to the numbering difference: \code{i2c-0},
+\code{i2c-1} is the Linux numbering, based on the registration order
+of enabled I2C busses. Here, because only I2C2 and I2C8 are enabled,
+they are called \code{i2c-0} and \code{i2c-1}.
+
+\begin{verbatim}
+# i2cdetect -r 1
+i2cdetect: WARNING! This program can confuse your I2C bus
+Continue? [y/N] y
+      0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
+00:          -- -- -- -- -- -- -- -- -- -- -- -- -- 
+10: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
+20: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
+30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
+40: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
+50: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
+60: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
+70: -- -- -- -- -- -- -- -- 
+\end{verbatim}
+
+No device is detected yet, because this bus is just
+used for external devices.
+
+Now try to probe I2C2 through \code{i2cdetect -r 0}.
+
+You will see that the command will run very slowly because of failing to
+connect to the bus. That's because the corresponding signals are
+not exposed yet to the outside connectors through pin muxing.
```

Comment:
I'm confused here because previously you disabled pin-muxing for I2C8... so I2C8 should exhibit this behavior as well. On the other hand you say earlier that I2C2 is already enabled/used. I'm rather confused.